src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -45,10 +52,18 @@ public SubstringConstraint(string expected) : base(expected)
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            return actual != null && actual.IndexOf(expected, comparisonType) >= 0;
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied StringComparison type.
```

Comment:
Since I've verified that this is the behavior that matches `EqualConstraint`, can the summary be `Overrides the <see cref="IgnoreCase"/> setting and instead uses the specified comparison.`?